internal/gcsx/random_reader.go

```diff
@@ -233,6 +233,16 @@ func (rr *randomReader) tryReadingFromFileCache(ctx context.Context,
 	n = 0
 
 	if cacheutil.IsCacheHandleInvalid(err) {
+		// Invalidate the cache for the object corresponding to async job. Next read call for the object
+		// will create a fresh async job and download the content if needed.
+		if strings.Contains(err.Error(), cacheutil.FailedFileDownloadJobErrMsg) {
```

Comment:
should we handle this in cache_handler.go ? here we should just close the cache handle that we are closing.

Reply:
Hi Ayush,

Created another PR for this approach - https://github.com/GoogleCloudPlatform/gcsfuse/pull/1663

I'll close this PR once another gets merged.